static/js/components/CompactCourseDisplay.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+/* global SETTINGS:false */
+import React from "react"
+import { connect } from "react-redux"
+
+import Card from "./Card"
+import { courseAvailability, maxPrice } from "../lib/courses"
+import { embedlyThumbnail } from "../lib/url"
+import { EMBEDLY_THUMB_HEIGHT, EMBEDLY_THUMB_WIDTH } from "../lib/posts"
+
+import type { Course } from "../flow/discussionTypes"
+
+type Props = {
+  course: Course
+}
+
+export class CompactCourseDisplay extends React.Component<Props> {
```

Comment:
It will be updating state in the subsequent PR for the course detail view:
```python
type Props = {
  course: Course,
  dispatch: Dispatch<*>
}

export class CompactCourseDisplay extends React.Component<Props> {
  setCourseForDrawer = async () => {
    const { dispatch, course } = this.props
    dispatch(setShowCourseDrawer({ visible: true, courseId: course.id }))
  }

  render() {
    const { course } = this.props
    return (
      <Card className={`compact-course-summary`}>
        <div className="column1" onClick={this.setCourseForDrawer}>
     .......

```